src/fetchcode/package.py

```diff
@@ -362,6 +376,290 @@ def get_gnu_data_from_purl(purl):
     )
 
 
+@router.route("pkg:cocoapods/.*")
+def get_cocoapods_data_from_purl(purl):
+    logging.basicConfig(
+        filename=LOG_FILE_LOCATION,
+        level=logging.WARN,
+        format="%(levelname)s - %(message)s",
+        filemode="w",
+    )
+    input_purl = purl
+    purl = PackageURL.from_string(purl)
+    name = purl.name
+    version = purl.version
+    cocoapods_org_url = f"https://cocoapods.org/pods/{name}"
+    repository_homepage_url = f"https://cocoapods.org/pods/{name}"
+
+    # This dictionary helped me monitor the values as I worked on the code.
+    # Only 2 key-value pairs are currently used below to define variables:
+    # `input_name` and `cocoapods_org_pod_name`.
+
+    pod_summary = {}
+    pod_summary["input_purl"] = input_purl
+    pod_summary["input_name"] = name
+    pod_summary["cocoapods_org_url"] = cocoapods_org_url
+    pod_summary["repository_homepage_url"] = repository_homepage_url
+    pod_summary["no_github_repo"] = None
+    pod_summary["gh_repo_four_o_four"] = None
+    pod_summary["http_url"] = None
+
+    cocoapods_org_url_head_request = make_head_request(cocoapods_org_url)
+    cocoapods_org_url_status_code = cocoapods_org_url_head_request.status_code
+    pod_summary["cocoapods_org_url_status_code"] = cocoapods_org_url_status_code
+    if cocoapods_org_url_status_code == 404:
+        logger.error(f"cocoapods_org_url not found for {name}")
+        return
+    elif cocoapods_org_url_status_code == 302:
+        redirect_url = cocoapods_org_url_head_request.headers['Location']
+        redirect_message = f"The cocoapods.org URL {cocoapods_org_url} redirects to {redirect_url}"
+        logger.error(redirect_message)
```

Comment:
Why is it an error if it's a redirect

Reply:
Thanks @TG1999 .  I'll change this to a `logger.warning`.  There may be a few others that also need "downgrading" from `logger.error` to `logger.warning` and I'll keep my eyes peeled for those.  And I'll be especially interested in your thoughts on the refactored cocoapods code and tests once I've finished that.  🙂